File: Samples/ARTCCM-TNIR-sample_EN_ver.1.3.1.3c/Sample/Win32/Delphi2006/Sample_Delphi_2006_FullControl/Unit4.pas
unit Unit4;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, ArtCam;

type
  TForm4 = class(TForm)
    GroupBox1: TGroupBox;
    GroupBox2: TGroupBox;
    CheckRead1: TCheckBox;
    CheckRead2: TCheckBox;
    CheckRead3: TCheckBox;
    CheckRead4: TCheckBox;
    CheckRead5: TCheckBox;
    CheckRead6: TCheckBox;
    CheckRead7: TCheckBox;
    CheckRead8: TCheckBox;
    CheckWrite1: TCheckBox;
    CheckWrite2: TCheckBox;
    CheckWrite3: TCheckBox;
    CheckWrite4: TCheckBox;
    CheckWrite5: TCheckBox;
    CheckWrite6: TCheckBox;
    CheckWrite7: TCheckBox;
    CheckWrite8: TCheckBox;
    Button1: TButton;
    Button2: TButton;
    Button3: TButton;
    procedure FormShow(Sender: TObject);
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure Button3Click(Sender: TObject);
  private
    { Private declaration }
  public
    m_pCArtCamSdk: ^TArtCam;
    m_DllType: ARTCAM_CAMERATYPE;
  end;

var
  Form4: TForm4;

implementation

{$R *.dfm}

procedure TForm4.FormShow(Sender: TObject);
begin
  if nil = m_pCArtCamSdk then begin
    exit;
  end;

  // Change operation for each DLL
  // For each DLL function, please refer to manual.
  m_DllType := ARTCAM_CAMERATYPE(HIWORD(m_pCArtCamSdk.GetDllVersion()));


  case m_DllType of
    ARTCAM_CAMERATYPE_130MI,
    ARTCAM_CAMERATYPE_200MI,
    ARTCAM_CAMERATYPE_300MI: begin

	 		Button1.Enabled := TRUE;
			CheckRead1.Enabled := TRUE;
			CheckRead2.Enabled := TRUE;
			CheckRead3.Enabled := TRUE;
			CheckRead4.Enabled := TRUE;
			CheckRead5.Enabled := TRUE;
			CheckRead6.Enabled := TRUE;
			CheckRead7.Enabled := TRUE;
			CheckRead8.Enabled := TRUE;

			Button2.Enabled := TRUE;
			CheckWrite1.Enabled := TRUE;
			CheckWrite2.Enabled := TRUE;
			CheckWrite3.Enabled := TRUE;
			CheckWrite4.Enabled := TRUE;
			CheckWrite5.Enabled := TRUE;
			CheckWrite6.Enabled := TRUE;
			CheckWrite7.Enabled := TRUE;
			CheckWrite8.Enabled := TRUE;
  end;

end;
end;

procedure TForm4.Button1Click(Sender: TObject);
begin
//  Read
end;

procedure TForm4.Button2Click(Sender: TObject);
begin
//  Write
end;

procedure TForm4.Button3Click(Sender: TObject);
begin
  Hide;
end;

end.

File: Samples/ARTCCM-TNIR-sample_EN_ver.1.3.1.3c/Sample/Win32/BCB2006/Sample_BCB_2006_FullControl/Unit4.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit4.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm4 *Form4;
//---------------------------------------------------------------------------
__fastcall TForm4::TForm4(TComponent* Owner)
    : TForm(Owner)
{
	m_pCArtCamSdk = NULL;
}
//---------------------------------------------------------------------------
void __fastcall TForm4::FormShow(TObject *Sender)
{
	// Change operation for each DLL
	// For each DLL function, please refer to manual.
	m_DllType = HIWORD(m_pCArtCamSdk->GetDllVersion());


	switch(m_DllType){
	case ARTCAM_CAMERATYPE_130MI:
	case ARTCAM_CAMERATYPE_200MI:
	case ARTCAM_CAMERATYPE_300MI:
		Button1->Enabled = TRUE;
		CheckRead1->Enabled = TRUE;
		CheckRead2->Enabled = TRUE;
		CheckRead3->Enabled = TRUE;
		CheckRead4->Enabled = TRUE;
		CheckRead5->Enabled = TRUE;
		CheckRead6->Enabled = TRUE;
		CheckRead7->Enabled = TRUE;
		CheckRead8->Enabled = TRUE;

		Button2->Enabled = TRUE;
		CheckWrite1->Enabled = TRUE;
		CheckWrite2->Enabled = TRUE;
		CheckWrite3->Enabled = TRUE;
		CheckWrite4->Enabled = TRUE;
		CheckWrite5->Enabled = TRUE;
		CheckWrite6->Enabled = TRUE;
		CheckWrite7->Enabled = TRUE;
		CheckWrite8->Enabled = TRUE;
	break;
	}

}
//---------------------------------------------------------------------------
void __fastcall TForm4::Button1Click(TObject *Sender)
{
	BYTE Data = 0x00;
	if(m_pCArtCamSdk->GetIOPort(&Data, NULL, 0)){
		CheckRead1->Checked = (Data & 0x01) ? TRUE : FALSE;
		CheckRead2->Checked = (Data & 0x02) ? TRUE : FALSE;
		CheckRead3->Checked = (Data & 0x04) ? TRUE : FALSE;
		CheckRead4->Checked = (Data & 0x08) ? TRUE : FALSE;
		CheckRead5->Checked = (Data & 0x10) ? TRUE : FALSE;
		CheckRead6->Checked = (Data & 0x20) ? TRUE : FALSE;
		CheckRead7->Checked = (Data & 0x40) ? TRUE : FALSE;
		CheckRead8->Checked = (Data & 0x80) ? TRUE : FALSE;
	}
}
//---------------------------------------------------------------------------
void __fastcall TForm4::Button2Click(TObject *Sender)
{
	BYTE Data = 0x00;
	if(CheckWrite1->Checked) Data |= 0x01;
	if(CheckWrite2->Checked) Data |= 0x02;
	if(CheckWrite3->Checked) Data |= 0x04;
	if(CheckWrite4->Checked) Data |= 0x08;
	if(CheckWrite5->Checked) Data |= 0x10;
	if(CheckWrite6->Checked) Data |= 0x20;
	if(CheckWrite7->Checked) Data |= 0x40;
	if(CheckWrite8->Checked) Data |= 0x80;

	m_pCArtCamSdk->SetIOPort(Data, 0, 0);

}
//---------------------------------------------------------------------------
void __fastcall TForm4::Button3Click(TObject *Sender)
{
	Hide();
}
//---------------------------------------------------------------------------

